src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -501,6 +501,29 @@ private TestExecutionContext CreateIsolatedContext()
             return context;
         }
 
+        /// <summary>
+        /// Invokes <paramref name="action"/> in this context instance, handling all exceptions.
+        /// </summary>
+        /// <param name="action">May be <see langword="null"/>.</param>
+        public void ExecuteWithExceptionHandling(Action<TestExecutionContext> action)
+        {
+            if (action == null) return;
+
+            try
+            {
+                action.Invoke(this);
+            }
+            catch (Exception ex)
+            {
+#if PARALLEL
+                if (ex is ThreadAbortException)
+                    Thread.ResetAbort();
+#endif
+                if (CurrentResult == null) CurrentResult = CurrentTest.MakeTestResult();
+                CurrentResult.RecordException(ex);
+            }
+        }
+
```

Comment:
I like the idea of such a method but this seems a tad too general for clarity. What if it took a work test command rather than an action.

Reply:
I'm open to ideas, but that would break the usage in BeforeAndAfterTestCommand:
https://github.com/nunit/nunit/pull/2565/files#diff-d7f76a7573ec9da3e9654efe7393eaddR57